doc/version_history.rst

```diff
@@ -6,6 +6,19 @@
 Version History
 ###############
 
+v0.17.1
```

Comment:
I think these are quite substantial changes for a patch release. Consider making this a minor release, e.g. `v0.18.0`